keel-core/src/main/kotlin/com/netflix/spinnaker/keel/core/api/ClusterDeployStrategy.kt

```diff
@@ -23,42 +23,90 @@ import java.time.Duration.ZERO
   Type(Highlander::class)
 )
 sealed class ClusterDeployStrategy {
-  abstract fun toOrcaJobProperties(): Map<String, Any?>
   @get:JsonIgnore
   open val isStaggered: Boolean = false
   @get:JsonInclude(NON_EMPTY)
   open val stagger: List<StaggeredRegion> = emptyList()
+  abstract fun toOrcaJobProperties(): Map<String, Any?>
+  abstract fun withDefaultsOmitted(): ClusterDeployStrategy
 }
 
 @JsonTypeName("red-black")
 data class RedBlack(
-  val rollbackOnFailure: Boolean = true,
-  val resizePreviousToZero: Boolean = false,
-  val maxServerGroups: Int = 2,
-  val delayBeforeDisable: Duration = ZERO,
-  val delayBeforeScaleDown: Duration = ZERO,
+  val rollbackOnFailure: Boolean? = true,
+  val resizePreviousToZero: Boolean? = false,
+  val maxServerGroups: Int? = 2,
+  val delayBeforeDisable: Duration? = ZERO,
```

Comment:
Why do these need to be nullable now?

Reply:
Sorry @robfletcher, I was still adding some additional inline comments: https://github.com/spinnaker/keel/pull/1055#discussion_r412556993